src/cider/nrepl.clj

```diff
@@ -12,6 +12,10 @@
   "Map of `delay`s holding deferred middleware handlers."
   (atom nil))
 
+(def LOCK
```

Comment:
I really think this should be named differently - LOCK is way too generic (and as it's not a constant I don't think it should all-caps either).

Reply:
I use caps to emphasize that those a global variables.  Some form of marking is needed to differentiate them from locals. All-caps is a useful marking technique and restricting it to only constants is a too massive of a restriction IMO. 

I can make them dynamic and use `*lock*` if that's what you are more comfortable with. But we will never use the dynamic nature of the vars then. 

As to the names, how about `REQUIRE-LOCK`?